### PR TITLE
Global Styles Sidebar: Re-add Colors: Heading to selected blocks

### PR DESCRIPTION
**Pull request by carolinan**
### WIP

Prior to https://github.com/WordPress/gutenberg/pull/48893,
The panel at `Site Editor Styles > Blocks > Block name > Colors` would show a heading color option for all blocks that supported text color.
This was incorrect, and solved in 48893.

This PR adds the Colors > Heading option back for selected blocks only, through opt-in in block.json.

## Why?

These blocks has or can have headings inside them, and having this option in the Styles sidebar is a useful feature.

## Testing Instructions
1. Activate a theme with support for the site editor.
2. Open the Styles sidebar in the Site Editor.
3. Confirm that root level Styles > Colors > Elements > Headings still work.
4. Select Styles > Blocks > Columns, and confirm that the Colors panel includes Headings.
5. Select Styles > Blocks > Verse, and confirm that the Colors panel do not include Headings.

## Screenshots or screencast 
Before:
The Verse block has a heading color option in the Styles sidebar:
https://user-images.githubusercontent.com/7422055/225591043-2cd43ce7-aaf1-487f-9c3c-0d3f0f8f2413.mov

After:
The Verse block does not have a heading color option in the Styles sidebar:
https://user-images.githubusercontent.com/7422055/233548549-683c8dd2-62ee-4831-a169-7526aebaf05d.mp4

### CODE REVIEW

File: packages/block-editor/src/components/global-styles/utils.js
Comment (MaggieCabrera) [inline packages/block-editor/src/components/global-styles/utils.js]: do you know why the button is missing `buttonBackgroundColor` here? could we use just the one?

File: packages/blocks/src/store/private-selectors.js
Comment (youknowriad) [inline packages/blocks/src/store/private-selectors.js]: For the button and link colors, we don't separate between, text and background colors, should we follow the same approach here for simplicity?

File: packages/block-editor/src/components/global-styles/utils.js
Comment (carolinan) [reply packages/block-editor/src/components/global-styles/utils.js]: By Riad's suggestion, I take it that using only one would work,
but I would have to refer to @scruffian if this is even the correct method to hide the options for the individual blocks.

**Comment by youknowriad** [comment]
FYI, in this PR https://github.com/WordPress/gutenberg/pull/48893 where I'm refactoring the color components in both global styles and block inspector. I had to solve this issue as well within the refactoring. I mapped the behavior of the "button" colors.

**Comment by carolinan** [comment]
@youknowriad Should this be closed then? I am not sure it includes any changes that would be needed on top of your refactor.

**Comment by youknowriad** [comment]
My PR is more complex, I didn't think initially that I would be able to push it to the finish line so soon but apparently, yeah, it's very close now.

**Comment by scruffian** [comment]
Thanks for working on this

> Are there any core blocks that need to opt-in? I don't think there are? The cover block and media & text blocks uses inner paragraphs, not headings.

The following blocks can contain headings:
- Quote
- Cover
- Media and Text
- Columns
- Group
- Post Content
- Comments
- Post Comments Form

I think we should enable all of this.

**Comment by youknowriad** [comment]
@scruffian do we have the block supports for this property? I mean if I set a custom heading color at the block level in a "style" attribute, do I see the result in the frontend?

If not, I think we should hold on adding support for these blocks (in block.json I mean) because with my PR unifying the behavior of global styles and block inspector, it will also show the UI in the block inspector but the behavior won't work.

**Comment by scruffian** [comment]
I'm happy to hold off until your PR merges, just stating what I think the ultimate behaviour should be, even if it takes a while to get there :)

**Comment by carolinan** [comment]
> do we have the block supports for this property?

To my understanding, not today, not without your PR or this PR.

I am not opposed to holding off this PR.

I am not opposed to implementing the heading option for the blocks mentioned:
But all of these blocks can **also** have button blocks (including the comments form which has one button), but we did not implement button color support for them?

**Comment by MaggieCabrera** [comment]
> I am not opposed to implementing the heading option for the blocks mentioned: But all of these blocks can **also** have button blocks (including the comments form which has one button), but we did not implement button color support for them?

This looks like an oversight to me or lack of time to go back to it and add the functionality in. I don't think it was intentional. I did the same thing with the captions and I need to follow up to include them on the blocks that can actually have them.

**Comment by carolinan** [comment]
Yes I believe the same. 
Let's follow up after Riad's PR has been reviewed and merged.

**Comment by carolinan** [comment]
Trying to remember what needs to be done as the next step. 
Note to self:
1) There may be too many conflicts to continue with this PR
2) With trunk, and updating the group block block.json to use:
```
	"color": {
			"gradients": true,
			"link": true,
			"caption": true,
			"button": true,
			"heading": true,
			"__experimentalDefaultControls": {
				"background": true,
				"text": true
			}
		},
```
Then caption and button show correctly in the Styles > Blocks > Group > Colors.

To do:
1. Enable headings
2. Opt-into headings in block.json of the blocks that Ben listed above.
3. Learn if any or which blocks should also opt-in to show button and caption.

File: packages/block-editor/src/components/global-styles/color-panel.js
Comment (aristath) [inline packages/block-editor/src/components/global-styles/color-panel.js:38]: Good catch! This was a typo/remnant from the older implementation that was never changed 👍

**Review by aristath — APPROVED** [review]
Great work @carolinan!

This is a simple change, the code looks good and works as expected.
LGTM 👍 

There is a conflict that needs to be resolved, but other than that this looks good to merge 🎉

**Review by scruffian — APPROVED** [review]
LGTM. Just noting it's the column block that has support, not columns.

**Pull request merged**
